src/NServiceBus.RavenDB/Internal/DocumentStoreInitializer.cs

```diff
@@ -38,7 +38,9 @@ internal IDocumentStore Init(ReadOnlySettings settings)
 
         void ApplyConventions(ReadOnlySettings settings)
         {
-            var idConventions = new DocumentIdConventions(docStore, settings.GetAvailableTypes(), settings.EndpointName());
+            var sagasEnabled = settings.GetOrDefault<bool>(typeof(Features.Sagas).FullName);
```

Comment:
I would actually make 2 features, one that depends on Sagas, and it adds the relevant mappings
and the other that depends on timeouts and adds the relevant mapping.

So this means that the "mappings dictionary" needs to be a singleton